app/models/concerns/pxe_loader_support.rb

```diff
@@ -15,14 +17,20 @@ module PxeLoaderSupport
   }.with_indifferent_access.freeze
 
   class_methods do
-    def all_loaders_map(precision = 'x64')
+    def all_loaders_map(precision = 'x64', tftp_proxy = "tftp_proxy")
+      port = Setting[:http_boot_port]
+      path = Setting[:http_boot_path]
       {
         "None" => "",
         "PXELinux BIOS" => "pxelinux.0",
         "PXELinux UEFI" => "pxelinux.efi",
         "Grub UEFI" => "grub/grub#{precision}.efi",
         "Grub2 UEFI" => "grub2/grub#{precision}.efi",
-        "Grub2 UEFI SecureBoot" => "grub2/shim#{precision}.efi"
+        "Grub2 UEFI SecureBoot" => "grub2/shim#{precision}.efi",
+        "Grub2 UEFI HTTP" => "http://#{tftp_proxy}:#{port}#{path}/grub2/grub#{precision}.efi",
+        "Grub2 UEFI HTTPS" => "https://#{tftp_proxy}:#{port}#{path}/grub2/grub#{precision}.efi",
+        "Grub2 UEFI HTTPS SecureBoot" => "https://#{tftp_proxy}:#{port}#{path}/grub2/shim#{precision}.efi",
+        "iPXE UEFI HTTP" => "http://#{tftp_proxy}:#{port}#{path}/ipxe-#{precision}.efi"
```

Comment:
I also add iPXE UEFI which simply loads `ipxe-x64.efi` which can be grabbed from here: http://boot.ipxe.org/

We don't ship this with Foreman or Satellite tho, users can still try this workflow if they like it.

Reply:
Just my initial impression after a quick skim: I have mixed feelings about using the tftp_proxy for building the URL.   

Back when the templates proxy was implemented, this approach was taken and it turned out to be problematic.  I realize this is a different use case, but I think the end goal would be to have a smart proxy plugin called HTTPBOOT, right? Should it be responsible for knowing the URL for HTTP and HTTPS boot and returning that to Foreman like template_url?

Alos, I realize this is the first "manual" approach.  The user is going to have to configure HTTP to serve the files.  Could we, at least for the initial deployment, use host parameters instead for the URL? Then once we implement a smart proxy plugin that's a little more intelligent use that.

Reply:
If you prefer having our own module, I can create one, let's do this properly from the day one. The only concern is security bugs, allowing others to read arbitrary files. History shows that many people failed, many times :-)

But yes, this is cleaner from the user and dev perspective, the only drawback is that we will have a feature "httpboot" dependent on other feature "tftp" and I am not sure if smart-proxy plugin API supports that.